### PR TITLE
Use google style docstrings (napoleon), optional book theme

### PR DESCRIPTION
**Pull request by amitschang**
The google style docstrings provided by napoleon extension are easier to write and read than the traditional ones, so use this by default. A popular theme is the book theme which has some nice features, adding option for that which we could take as default if desired.

FYI, screenshots showing book theme and rtd theme:

**RTD**

image

**Book**

image

### CODE REVIEW

**Review by xiangchenjhu — APPROVED** [review]

File: docs/source/conf.py [hunk 1]
```diff
@@ -29,10 +31,22 @@
 
 # -- Options for HTML output
 
+# select theme from options below
 html_theme = 'sphinx_rtd_theme'
+#html_theme = 'sphinx_book_theme'
```
Comment (jamienoss) [inline docs/source/conf.py:36]: Should this be commented out?

Comment (amitschang) [reply docs/source/conf.py:36]: yeah, was meant to be an alternate choice of theme

**Comment by jamienoss** [comment]
@amitschang My retention for building docs is always short lived, so help me out here - is this a two in one PR or just the theme change?

Your description describes using the napolean style as it's easier to write for, so I see that this PR adds that to the deps, but it also looks like this PR also makes the switch to using the new theme as spec'd in the ``pyproject.toml`` - is this correct?

Just want to make sure I understand.

I have to confess I personally prefer the RTD theme as it's what I'm more used to seeing - but familiarity may b the only rationale/bias - but am happy to not care and make the change either way.

**Comment by amitschang** [comment]
The napoleon extension allows simpler docstrings, and that is the only thing that has a substantive effect for this (actually, I also added the viewcode which is handy).

The book theme is just setting up the dependency and default settings that are relevant to that so we can use it if desired in derivative projects.

**Review by jamienoss — APPROVED** [review]

**Comment by amitschang** [comment]
thanks!

**Pull request merged**